#goal definition
char[] inputData
---
#result definition
char[] data
---
#feedback
int32[] sequence